src/AdoNetCore.AseClient/AseException.cs

```diff
@@ -29,21 +30,27 @@ public AseErrorCollection Errors
             get; private set;
         }
 
+        public AseError MostSevereError
```

Comment:
I reckon you could store this information (maybe an index to the most severe error) in `AseErrorCollection`. In fact, since the collection can't be modified after construction, the most severe error could be determined when the collection is constructed.

This would avoid having to set `MostSevereError` on every `AseException` constructor call, and instead you could do something like `Message => Errors.Count == 0 ? string.Empty : Errors.MostSevereMessage`